parser_test.go

```diff
@@ -351,6 +351,54 @@ func (s *testParserSuite) RunRestoreTest(c *C, sourceSQLs, expectSQLs string) {
 	c.Assert(restoreSQLs, Equals, expectSQLs, comment)
 }
 
+func (s *testParserSuite) RunTestInRealAsFloatMode(c *C, table []testCase) {
+	parser := parser.New()
+	parser.EnableWindowFunc(s.enableWindowFunc)
+	parser.SetSQLMode(mysql.ModeRealAsFloat)
+	for _, t := range table {
+		_, _, err := parser.Parse(t.src, "", "")
+		comment := Commentf("source %v", t.src)
+		if !t.ok {
+			c.Assert(err, NotNil, comment)
+			continue
+		}
+		c.Assert(err, IsNil, comment)
+		// restore correctness test
+		if t.ok {
+			s.RunRestoreTestInRealAsFloatMode(c, t.src, t.restore)
+		}
+	}
+}
+
+func (s *testParserSuite) RunRestoreTestInRealAsFloatMode(c *C, sourceSQLs, expectSQLs string) {
```

Comment:
Is this still needed?

Reply:
I think that it is needed.  
It can judge whether `REAL_AS_FLOAT` has any effect on `CAST(… AS FLOAT)`
And, if it will support `CAST(… AS REAL)` in future , it also can use the function to test.